pip/pip-316.md

```diff
@@ -0,0 +1,83 @@
+
+# PIP-316: Create a producerName field for DeadLetterPolicy
+
+
+# Motivation
+
+Add producerName for dead letter producer, easier to locate problems.
+
+
+# Goals
+
+## In Scope
+
+Add a new field `producerName` in `org.apache.pulsar.client.api.DeadLetterPolicy` class.
+
+
+## Out of Scope
+None.
+
+# High Level Design
+
+If `producerName` is not empty, use `producerName` as the name of the dead letter producer.
```

Comment:
Could we use something like `${topic-name}-${subscription-name}-DLQ` as the producer name without introducing the `producerName` to the DLQ policy?

Reply:
That's a good suggestion, it can reduce the psychological burden on users when naming.

Reply:
Does this work for your case? So that we don't need to add a public interface for it.

Reply:
Yeah, it work for me, which name would be better?
```java
producerName: persistent://my-property/my-ns/dead-letter-topic-my-subscription-DLQ
producerName: my-property/my-ns/dead-letter-topic-my-subscription-DLQ
producerName: dead-letter-topic-my-subscription-DLQ
```

Reply:
I would prefer `persistent://my-property/my-ns/dead-letter-topic-my-subscription-DLQ`

Reply:
I have created a new PR https://github.com/apache/pulsar/pull/21589 , PTAL.